invenio_app_ils/ui/src/invenio_app_ils/pages/frontsite/Home/components/MostLoanedDocuments/MostLoanedDocuments.js

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { generatePath } from 'react-router';
+import { Loader, Error } from '../../../../../common/components';
+import { itemSearchQueryUrl, FrontSiteURLS } from '../../../../../common/urls';
+import { Button, Card, Divider } from 'semantic-ui-react';
+import { DocumentCard } from '../DocumentCard';
+import { shortenString } from '../../../../../common/helpers';
+
+export default class MostLoanedDocuments extends Component {
+  constructor(props) {
+    super(props);
+    this.fetchMostLoanedDocuments = props.fetchMostLoanedDocuments;
+    this.showDetailsUrl = FrontSiteURLS.bookDetails;
+    this.maxItemsToDisplay = props.maxDisplayedItems;
+  }
+
+  componentDidMount() {
+    this.fetchMostLoanedDocuments();
+  }
+
+  _showDetailsHandler = document_pid =>
+    this.props.history.push(
+      generatePath(this.showDetailsUrl, { documentPid: document_pid })
+    );
+
+  goTo = path => {
+    this.props.history.push(path);
+  };
+
+  prepareData(data) {
+    return data.hits
+      .slice(0, this.maxItemsToDisplay)
+      .filter(item => item.metadata.document_types.includes(this.props.filter))
+      .map(item => {
+        return {
+          pid: item.metadata.document_pid,
+          title: item.metadata.title,
+          authors: shortenString(item.metadata.authors.join('\n')),
+          imageSize: 'small',
+          imageCover: 'https://assets.thalia.media/img/46276899-00-00.jpg',
+          onClick: this._showDetailsHandler.bind(
+            this,
+            item.metadata.document_pid
+          ),
+        };
+      });
+  }
+
+  renderCards(items) {
+    const cards = items.map(item => {
+      return <DocumentCard key={item.pid} documentData={item} />;
+    });
+
+    return <Card.Group>{cards}</Card.Group>;
+  }
+
+  render() {
+    const { data, isLoading, error } = this.props;
+    const items = this.prepareData(data);
+    return (
+      <Loader isLoading={isLoading}>
+        <Error error={error}>
+          <div>
+            <Button
+              onClick={this.goTo.bind(
+                this,
```

Comment:
Are you sure that the bind here is needed? Because I think the arrow functions are by default binded to the class instance?

Reply:
There are two ways, but I chose the second one because it looked cleaner to me:

`<button onClick={(e) => this.handleClick(id, e)}>Handle Click</button>
<button onClick={this.handleClick.bind(this, id)}>Handle Click</button>`

Should I change it?

Reply:
Yes please, we try to use `arrow` functions to avoid to `bind` all the time :)

Reply:
No actually I thought you did it for the `thisArg` binding. Actually I didn't know that and indeed seems way cleaner :) Probably we need to change also the other handlers to this format :)

Reply:
I changed it so that it won't re-render stuff unnecessarily :)